sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client_helpers.py

```diff
@@ -0,0 +1,151 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=docstring-keyword-should-match-keyword-only
+
+from typing import (
+    Any, Dict, Optional, Tuple, Union,
+    TYPE_CHECKING
+)
+from urllib.parse import quote, unquote, urlparse
+
+from ._serialize import get_access_conditions, get_source_conditions
+from ._shared.base_client import parse_query
+from ._shared.response_handlers import return_response_headers
+
+if TYPE_CHECKING:
+    from urllib.parse import ParseResult
+
+
+def _parse_url(account_url: str, share_name: str, file_path: str) -> "ParseResult":
+    try:
+        if not account_url.lower().startswith('http'):
+            account_url = "https://" + account_url
+    except AttributeError as exc:
+        raise ValueError("Account URL must be a string.") from exc
+    parsed_url = urlparse(account_url.rstrip('/'))
+    if not (share_name and file_path):
+        raise ValueError("Please specify a share name and file name.")
+    if not parsed_url.netloc:
+        raise ValueError(f"Invalid URL: {account_url}")
+    return parsed_url
+
+
+def _parse_snapshot(
+    snapshot: Optional[Union[str, Dict[str, Any]]] = None,
+    path_snapshot: Optional[str] = None
+) -> Optional[str]:
+    if hasattr(snapshot, 'snapshot'):
+        return snapshot.snapshot
+    if isinstance(snapshot, Dict):
+        return snapshot['snapshot']
+    return snapshot or path_snapshot
+
+
+def _from_file_url(
+    file_url: str,
+    snapshot: Optional[Union[str, Dict[str, Any]]] = None
+) -> Tuple[str, str, str, Optional[Union[str, Dict[str, Any]]]]:
+    try:
+        if not file_url.lower().startswith('http'):
+            file_url = "https://" + file_url
+    except AttributeError as exc:
+        raise ValueError("File URL must be a string.") from exc
```

Comment:
Let's replace this try/catch with an `isinstance(file_url, str)` and throw in the else.